components/fxa-client/src/send_tab.rs

```diff
@@ -64,7 +65,14 @@ impl FirefoxAccount {
                 }
             };
         let encrypted_payload: EncryptedSendTabPayload = serde_json::from_value(payload)?;
-        let payload = encrypted_payload.decrypt(&send_tab_key)?;
-        Ok(IncomingDeviceCommand::TabReceived { sender, payload })
+        match encrypted_payload.decrypt(&send_tab_key) {
+            Ok(payload) => Ok(IncomingDeviceCommand::TabReceived { sender, payload }),
+            Err(e) => {
```

Comment:
How much additional work would it be to re-fetch our own device record and check whether its contents match what we expect? I have to assume that the existing device registration is somehow incorrect, since re-registering fixes the problem. So this might be a good opportunity to try to log more diagnostics about exactly *how* it was incorrect.

Reply:
In my case everything looked all right (send tab key and push reg in order) and I never really figured why my keys were wrong. I'm not quite sure how could we compare the device contents since the random IV makes a comparison impossible :/
I believe we were hitting an error in this general area: https://searchfox.org/mozilla-mobile/rev/fd32c1b030795f42f68da92c36e9bdef97b52d29/application-services/components/fxa-client/src/send_tab.rs#56-67

Reply:
I think we'd have to try to decrypt our own send-tab key bundle and check that it matches what we expect from the plaintext.

Reply:
Added `diagnose_remote_keys`